barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/ultra_circuit_builder.cpp

```diff
@@ -417,12 +409,20 @@ template <typename FF> void UltraCircuitBuilder_<FF>::create_ecc_add_gate(const
     can_fuse_into_previous_gate = can_fuse_into_previous_gate && (q_4[this->num_gates - 1] == 0);
     can_fuse_into_previous_gate = can_fuse_into_previous_gate && (q_1[this->num_gates - 1] == 0);
     can_fuse_into_previous_gate = can_fuse_into_previous_gate && (q_arith[this->num_gates - 1] == 0);
+    can_fuse_into_previous_gate = can_fuse_into_previous_gate && (q_m[this->num_gates - 1] == 0);
```

Comment:
Coming back to this: this implies that I cannot fuse a double and then an addition right?

Reply:
Okay my mistake was not remembering that in the doubling gate we have:
```
    if (can_fuse_into_previous_gate) {
        q_elliptic[this->num_gates - 1] = 1;
        q_m[this->num_gates - 1] = 1; // This will switch it to a doubling gate
```